src/utils/populate.utils.ts

```diff
@@ -0,0 +1,121 @@
+import {mkdirSync} from 'fs';
+import {readFile} from 'fs/promises';
+import {red} from 'kleur';
+import {existsSync} from 'node:fs';
+import {mkdir, readdir, writeFile} from 'node:fs/promises';
+import {dirname, join, relative, resolve} from 'node:path';
+import ora from 'ora';
+import type {GitHubTreeEntry} from '../types/github';
+
+interface FileDescriptor {
+  path: string;
+  url: string;
+}
+
+const REPOSITORY = 'junobuild/create-juno';
+
+const getLocalFiles = async (templatePath: string): Promise<FileDescriptor[]> => {
+  const dirents = await readdir(templatePath, {withFileTypes: true});
+  const files = await Promise.all(
+    dirents.map(async (dirent) => {
+      const res = resolve(templatePath, dirent.name);
+      return dirent.isDirectory() ? await getLocalFiles(res) : res;
+    })
+  );
+  return files
+    .flat()
+    .map((p) =>
+      typeof p === 'string' ? {path: relative(process.cwd(), p), url: `file://${p}`} : p
+    );
+};
+
+const getGitHubFiles = async (templatePath: string): Promise<FileDescriptor[]> => {
+  const res = await fetch(`https://api.github.com/repos/${REPOSITORY}/git/trees/main?recursive=1`);
+  const {tree, message}: {tree: GitHubTreeEntry[]; message?: string} = await res.json();
+
+  if (tree === undefined) {
+    console.log(
+      `${red(`No files can be found on GitHub.${message !== undefined ? ` ${message}` : ''}`)}`
+    );
+    process.exit(1);
+  }
+
+  return tree
+    .filter(({type, path: p}) => type === 'blob' && p.startsWith(templatePath))
+    .map(({path}) => ({
+      path,
+      url: `https://raw.githubusercontent.com/${REPOSITORY}/main/${path}`
+    }));
+};
+
+const createDirectory = async (where: string | null) => {
+  if (where === null) {
```

Comment:
where can be null? should the user specify a destination or null === same directory?

Reply:
Yes, null implies the user chose the current directory as the destination, either by pressing Enter at the project name prompt (no input), or typing '.' (current directory).

There are two possible cases:
1. The user alread has created a destination directory, is inside it, and runs the CLI to populate (what I usually do)
2. The user lets the CLI create a directory for them, by naming it.

The "failure" cases here would be:
1. The CLI creates another subdirectory with the same name (resulting in `path/to/foo/foo/templateFilesHere`)
2. The CLI populates the CWD which is non-empty

Suggesting the CWD as a placeholder project name solves 1., to solve 2. we'd need to detect if the target directory exists and isn't empty.

What do you think?

Reply:
Thanks for the explanation. Let's start with the existing and iterate with some tests if needed.